README.md

```diff
@@ -206,22 +206,19 @@ between vertices $i$ and $j$ on the cortical sheet
-    + `sinkhorn`: extension of Sinkhorn algorithm, called "scaling algorithm" in [(Chizat et al. 2016) [2]](#2)
-    + `mm`: majorization-minimization algorithm in [(Chapel et al. 2021) [4]](#4)
-    + `dc`: unbalanced extension of inexact Bregman proximal point algorithm in [(Xie et al. 2020) [5]](#5)
-* `nits_bcd`: integer, number of BCD iterations to run
-* `nits_uot`: integer, number of iterations to solve the unbalanced optimal transport problems within each BCD iteration
-* `tol_bcd`: float, stop the BCD procedure if the absolute difference between two consecutive transport plans is under this threshold
-* `tol_uot`: float, stop the UOT procedure in each BCD iteration if the absolute difference between two consecutive dual vectors or marginal distributions (depending on the solver) is under this threshold
-* `early_stopping_threshold`: float, stop the BCD procedure early if the FUGW loss falls under this threshold
-* `eval_bcd`: integer. During .fit(), evaluate the FUGW cost at the multipliers of iteration. For example, if `eval_bcd=7`, then the FUGW cost is calculated at iterations 7, 14, 21, 28, etc...
-* `eval_uot`: integer. During .fit(), evaluate the absolute difference between two consecutive dual vectors or marginal distributions (depending on the solver) at the multipliers of iteration. For example, if `eval_uot=7`, then the evaluation happens at iterations 7, 14, 21, 28, etc...
-* `dc_eps_base`: float, Regularization parameter specific to the `"dc"` solver. It is recommended that it should not be neither too small nor too large.
-* `dc_nits_sinkhorn`: integer, Number of Sinkhorn iterations to run within each iteration of the `"dc"` solver
-* `device`: string, `torch.device` on which computation should happen
+* `init_plan`: array of size `(n, m)`
+* `init_duals`: tuple of arrays of size `(n)` and `(m)` respectively
+* `uot_solver`: `"sinkhorn"` or `"mm"` or `"ibpp"`
+* `nits_bcd`: number of BCD iterations to run
+* `nits_uot`: number of solver iterations to run within each BCD iteration
+* `tol_bcd`: Stop the BCD procedure early if the absolute difference between two consecutive transport plans under this threshold
+* `tol_uot`: Stop the BCD procedure early if the absolute difference between two consecutive transport plans under this threshold
+* `early_stopping_threshold`: Stop the BCD procedure early if the FUGW loss falls under this threshold
+* `eval_bcd`: During .fit(), at every eval_bcd step: 1. compute the FUGW loss and store it in an array 2. consider stopping early
+* `eval_uot`: During .fit(), at every eval_uot step: 1. consider stopping early
+* `ibpp_eps_base`: Regularization parameter specific to the ibpp solver
+* `ibpp_nits_sinkhorn`: Number of sinkhorn iterations to run within each uot iteration of the ibpp solver
+* `device`: `torch.device` on which computation should happen
```

Comment:
do we need to apss the device? I mean isn't it passed implicitly with the data?

Reply:
Input data can be in `np.ndarray` format, in which case it was useful for me to specify the GPU (in case other GPUs of the cluster are being used).
But maybe we could make a torch-only API, which would actually be easier to maintain.

Reply:
you mena if it is in numpy you convert it to torch?

Reply:
Yes

Reply:
I it makes sens but then it should be stated in the documentation (that devices is here/works for numpy input)

Reply:
I'll probably address this point in another PR and merge this one for now :)

Reply:
of course